tests/proxy/map_paging_predicate_test.py

```diff
@@ -0,0 +1,241 @@
+import os
+import unittest
+
+from tests.base import HazelcastTestCase
+from tests.util import configure_logging, get_abs_path, random_string
+from tests.custom_comparator import CustomComparator
+from hazelcast.serialization.predicate import PagingPredicate, is_greater_than_or_equal_to, is_less_than_or_equal_to, \
+    is_ilike
+from hazelcast import HazelcastClient
+from hazelcast.six import assertCountEqual
+from hazelcast.util import ITERATION_TYPE
+
+
+class MapPagingPredicateTest(HazelcastTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        configure_logging()
+
+    def setUp(self):
+        self.rc = self.create_rc()
+        self.cluster = self.create_cluster(self.rc, self._configure_cluster())
+        self.member1 = self.cluster.start_member()
+        self.member2 = self.cluster.start_member()
+        self.client = HazelcastClient()
+        self.map = self.client.get_map(random_string())
+
+    def tearDown(self):
+        self.map.destroy()
+        self.client.shutdown()
+        self.rc.exit()
+
+    """
+    Tests for proxy: comparator None
+    """
+    # Time out for assertTrueEventually set to 30. Could it be shorter?
+    def test_entry_set_with_paging_predicate(self):
+        self._fill_map_simple()
+        entry_set = self.map.entry_set(PagingPredicate(is_greater_than_or_equal_to('this', 3), 1)).result()
+
+        def assert_event():
+            self.assertEqual(len(entry_set), 1)
+            assertCountEqual(self, entry_set[0], ['key-3', 3])
+
+        self.assertTrueEventually(assert_event)
+
+    def test_key_set_with_paging_predicate(self):
+        self._fill_map_simple()
+        key_set = self.map.key_set(PagingPredicate(is_greater_than_or_equal_to('this', 3), 1)).result()
+
+        def assert_event():
+            self.assertEqual(len(key_set), 1)
+            self.assertEqual(key_set[0], 'key-3')
+
+        self.assertTrueEventually(assert_event)
+
+    def test_values_with_paging_predicate(self):
+        self._fill_map_simple()
+        values = self.map.values(PagingPredicate(is_greater_than_or_equal_to('this', 3), 1)).result()
+
+        def assert_event():
+            self.assertEqual(len(values), 1)
+            self.assertEqual(values[0], 3)
+
+        self.assertTrueEventually(assert_event)
+
+    """
+    Tests for paging: comparator None
+    """
+    def test_first_page(self):
+        self._fill_map_numeric()
+        paging = PagingPredicate(is_greater_than_or_equal_to('this', 40), 2)
+        self.assertTrueEventually(lambda: assertCountEqual(self, self.map.values(paging).result(), [40, 41]))
```

Comment:
Can you use closure instead of lambda here and below (like you did above). They are much more readable